app/presenters/account_show_presenter.rb

```diff
@@ -82,6 +82,21 @@ def formatted_legacy_idv_date
     I18n.l(user.active_profile.created_at, format: :event_date)
   end
 
+  def connect_to_initiating_idv_sp_url
+    initiating_service_provider = user.active_profile&.initiating_service_provider
+    return nil if !initiating_service_provider.present?
+
+    SpReturnUrlResolver.new(service_provider: initiating_service_provider).post_idv_follow_up_url
+  end
+
+  def connected_to_initiating_idv_sp?
+    initiating_service_provider = user.active_profile&.initiating_service_provider
+    return false if !initiating_service_provider.present?
+
+    identity = user.identities.find_by(service_provider: initiating_service_provider.issuer)
+    !!identity&.last_ial2_authenticated_at.present?
```

Comment:
Not sure we have test coverage for the scenario that `identity` is `nil` here (tests pass with nil-safe removed). What would the scenario be for that to happen?

Reply:
I was looking at the SAML flow and it appears that does not create an identity until right before redirect. In that case a user may not have an identity if a service provider does not do a step up flow.

You are right that we should have coverage for a nil identity scenario. Let me add that.